service/src/main/java/uk/nhs/adaptors/gp2gp/gpc/GetGpcDocumentReferencesTaskExecutor.java

```diff
@@ -0,0 +1,123 @@
+            ehrExtractStatusService.updateEhrExtractStatusAccessDocumentDocumentReferences(taskDefinition, urls);
+
+            urls.forEach(url -> queueGetDocumentsTask(taskDefinition, url));
+        }
+
+    }
+
+    private String retrievePatientId(GetGpcDocumentReferencesTaskDefinition taskDefinition) {
+        var request = gpcRequestBuilder.buildGetPatientIdentifierRequest(taskDefinition);
+        var response = gpcClient.getPatientRecord(request, taskDefinition);
+
+        FhirContext ctx = FhirContext.forDstu3();
+        IParser parser = ctx.newJsonParser();
+
+        if (response.contains("Bundle")) {
+            Bundle fhirBundle = parser.parseResource(Bundle.class, response);
+            if (!fhirBundle.getEntry().isEmpty()) {
+                Patient patient = (Patient) fhirBundle.getEntry().get(0).getResource();
+                return patient.getIdElement().getIdPart();
+            }
+        }
+        return StringUtils.EMPTY;
+    }
+
+    private List<String> retrieveDocumentReferences(GetGpcDocumentReferencesTaskDefinition taskDefinition, String patientId) {
+        var request = gpcRequestBuilder.buildGetPatientDocumentReferences(taskDefinition, patientId);
+        var response = gpcClient.getDocumentReferences(request, taskDefinition);
+
+        if (response.contains("Bundle")) {
+            FhirContext ctx = FhirContext.forDstu3();
+            IParser parser = ctx.newJsonParser();
+
+            Bundle bundle = parser.parseResource(Bundle.class, response);
+            return bundle.getEntry()
+                .stream()
+                .filter(fr -> fr.getResource().getResourceType().equals(ResourceType.DocumentReference))
+                .map(resource -> extractUrl((DocumentReference) resource.getResource()))
+                .filter(StringUtils::isNotEmpty)
+                .collect(Collectors.toList());
+        } else {
+            return new ArrayList<>();
+        }
+    }
+
+    private void queueGetDocumentsTask(GetGpcDocumentReferencesTaskDefinition taskDefinition, String url) {
+        var getGpcDocumentTaskTaskDefinition = GetGpcDocumentTaskDefinition.builder()
+            .documentId(GetGpcDocumentTaskDefinition.extractIdFromUrl(url))
+            .taskId(taskDefinition.getTaskId())
+            .conversationId(taskDefinition.getConversationId())
+            .requestId(taskDefinition.getRequestId())
+            .toAsid(taskDefinition.getToAsid())
+            .fromAsid(taskDefinition.getFromAsid())
+            .fromOdsCode(taskDefinition.getFromOdsCode())
+            .accessDocumentUrl(url)
+            .build();
+
+        taskDispatcher.createTask(getGpcDocumentTaskTaskDefinition);
+    }
+
+    private static String extractUrl(DocumentReference documentReference) {
+        if (documentReference.getContent().size() > 0) {
+            DocumentReference.DocumentReferenceContentComponent content = documentReference.getContent().get(0);
+            if (content.getAttachment() != null) {
```

Comment:
We can first get attachment and then create a single condition `if (attachment != null && !attachment.getUrl().isBlank())`

Reply:
And move it to another boolean method and name it `isUrlPresent()`.